File: src/main/java/com/yoghurt1131/weatherapi/application/controller/ForecastController.java
package com.yoghurt1131.weatherapi.application.controller;

import com.yoghurt1131.weatherapi.application.exception.ApiCallException;
import com.yoghurt1131.weatherapi.application.service.WeatherApiService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/forecast")
public class ForecastController {

    private final WeatherApiService weatherApiService;

    @Autowired
    public ForecastController(WeatherApiService weatherApiService) {
        this.weatherApiService = weatherApiService;
    }


    @GetMapping("/daily")
    public String dailyForecast() throws ApiCallException {
        weatherApiService.getTodaysWeather("Tokyo");

        return "OK";
    }
}

File: src/main/kotlin/dev/yoghurt1131/weatherapi/controller/ForecastController.kt
package dev.yoghurt1131.weatherapi.controller

import dev.yoghurt1131.weatherapi.application.service.WeatherApiService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/forecast")
class ForecastController(private val weatherApiService: WeatherApiService) {

    @GetMapping("/daily")
    fun dailyForecast(cityName: String) = weatherApiService.getTodaysWeather(cityName);

}